### PR TITLE
Define FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION when appropriate

### PR DESCRIPTION
**Pull request by sboukortt**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -231,12 +231,6 @@ if(MSVC)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS)
 endif()
 
-if("${JPEGXL_ENABLE_FUZZERS}" OR "${JPEGXL_ENABLE_COVERAGE}")
```
Comment (veluca93) [inline CMakeLists.txt:234]: Was this removal on purpose?

Comment (sboukortt) [reply CMakeLists.txt:234]: Yes, a quick `git grep` suggests that we don’t actually use it anywhere, and it is doubtful that we should.

**Review by veluca93 — DISMISSED** [review]

**Review by veluca93 — APPROVED** [review]
